feat : 문자열 덧셈 계신기

monkeyDugi opened the pull request.

monkeyDugi committed:
feat : 문자열 덧셈 계신기

monkeyDugi committed:
chore : 사용하지 않는 import 제거

lkimilhol commented on src/main/java/StringAddCalculator.java:
```diff
@@ -0,0 +1,59 @@
+    public static int sum(String text) {
+        int result = 0;
+
+        int[] numbers = splitToInt(text);
```
배열을 사용하신 이유가 있나요?

monkeyDugi replied:
크기가 명확하기 때문에 사용하였습니다. 물론 성능상 이슈는 전혀 없겠지만 ArrayList를 꼭 사용할 이유를 못찾았습니다.
아니면 반대로 특별한 이유가 없으면 ArrayList를 사용하는게 좋을까요?

lkimilhol replied:
정답은 없을거같습니다! 
말씀하신대로 크기가 명확하니 사용할수도 있겠지만 실수 할 여지는 분명히 있어보여요! (무슨 이유이던 간에 ㅎㅎ)
저라면 ArrayList를 사용하는 일급 컬렉션을 만들어서 2가지 숫자만 담기도록 하고 그렇지 않을 경우에는 생성이 되지 않도록 할거같네요!
(물론 정답은 절대 아닙니다!)

monkeyDugi replied:
일급 컬렉션으로 빼보려고 했는데 진행 하다보니 조금 과한 것 같아 그대로 유지하기로 결정 했습니다.
정답은 없다고 하셔서 저의 생각대로 결정을 했는데 어떨까요?

lkimilhol replied:
이번 again의 목적이 로또이니 추후 로또 개발 하시면서 해보시면 좋을거같아요!

lkimilhol commented on src/main/java/StringAddCalculator.java:
```diff
@@ -0,0 +1,59 @@
+        int[] numbers = new int[textLength];
+
+        for (int i = 0; i < textLength; i++) {
+            numbers[i] = Integer.parseInt(text[i]);
```
양수임을 보장하는 Positive 클래스를 만들면 어떨까요?

lkimilhol commented on src/main/java/StringAddCalculator.java:
```diff
@@ -0,0 +1,59 @@
+
+    private static final int CUSTOM_DELIMITER_INDEX = 1;
+    private static final int CUSTOM_DELIMITER_VALUE_INDEX = 2;
+    private static final String CUSTOM_DELIMITER_PATTERN = "//(.)\n(.*)";
```
static 변수들의 네이밍이 와닿지가 않는거같아요. 어떻게 생각하시나요?

lkimilhol commented on src/main/java/StringAddCalculator.java:
```diff
@@ -0,0 +1,59 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
```
유틸성으로 작성하기 위해 static 메서드를 제공하는 것이 맞을까요?

monkeyDugi replied:
과정에서 제공 받은 소스가 이렇게 되어 있어서 그대로 유지 하였습니다.
피드백의 의도는 static 메서드로 제공하는 것과 인스턴스 메서드로 제공하는 것에 대해 고민 해보라는 의미겠죠?

lkimilhol replied:
네네! 맞습니다! 고민은 충분히 잘 하셨으리라 생각합니다 :)

monkeyDugi replied:
단순히 덧셈 값을 반환만 받으면 되기 떄문에 상태를 가질 필요가 없기 때문입니다.

lkimilhol commented on src/test/java/StringAddCalculatorTest.java:
```diff
@@ -0,0 +1,48 @@
+        int result = StringAddCalculator.sum(null);
+        assertThat(result).isZero();
+
+        result = StringAddCalculator.sum("");
```
result가 아닌 새로운 변수를 넣어주면 어떨까요? 더 나아가 변수 자체를 선언하지 않으면 어떨까요?
왜 변수를 새로이 만들어 주는것이 나을지, 왜 변수를 선언하지 않으면 나을지 고민해보면 좋을거같아요!

lkimilhol reviewed (commented):
부족하지만 코멘트 달아보았어요.

monkeyDugi committed:
refactor : 클래스 분리
- Positive : 양수 배열을 생성하는 역할
- Delimiter : 문자열을 구분자로 분리하여 문자열 배열을 생성하는 역할

monkeyDugi committed:
refactor : 변수 네이밍 변경
[리뷰 링크](https://github.com/monkeyDugi/again-lotto/pull/1#discussion_r762400838)

monkeyDugi committed:
refactor : 의미없는 변수 제거
[리뷰 링크](https://github.com/monkeyDugi/again-lotto/pull/1#discussion_r762401231)
